File: Iniciante/1040 - Média 3/Python/1040.py
def main():
    entradas = input().split()
    notas = list(map(float, entradas))
    media = calculo_media(notas)
    print(f"Media: {media:.1f}")
    status = avaliacao_inicial(media)
    if status != "Aluno em exame.":
        print(status)
    else:
        print(status)
        exame(media)


def calculo_media(notas):
    return (notas[0] * 2 + notas[1] * 3 + notas[2] * 4 + notas[3]) / 10


def avaliacao_inicial(media):
    if media >= 7:
        return "Aluno aprovado."
    elif media < 5:
        return "Aluno reprovado."
    else:
        return "Aluno em exame."


def exame(media):
    nota_exame = float(input())
    print(f"Nota do exame: {nota_exame:.1f}")
    nova_media = (media + nota_exame) / 2
    if nova_media >= 5:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print(f"Media final: {nova_media:.1f}")


main()

File: Iniciante/1040 - Média 3/JavaScript/1040.js
function main() {
  let input = require("fs").readFileSync("/dev/stdin", "utf8");
  let lines = input.split('\n');

  const notas = lines[0].split(" ").map((item) => parseFloat(item, 10));
  const media = calculoMedia(notas);
  console.log(`Media: ${media.toFixed(1)}`);
  let situacao = avaliacaoInicial(media);

  if (situacao !== "Aluno em exame.") {
    console.log(situacao);
  } else {
    console.log(situacao);
    let notaExame = parseFloat(lines[1], 10);
    exame(media, notaExame);
  }
}

function calculoMedia(notas) {
  return (notas[0] * 2 + notas[1] * 3 + notas[2] * 4 + notas[3]) / 10;
}

function avaliacaoInicial(media) {
  if (media >= 7) {
    return "Aluno aprovado.";
  } else if (media < 5) {
    return "Aluno reprovado.";
  } else {
    return "Aluno em exame.";
  }
}

function exame(media, notaExame) {
  console.log(`Nota do exame: ${notaExame.toFixed(1)}`);
  let novaMedia = (media + notaExame) / 2;
  if (novaMedia >= 5) {
    console.log("Aluno aprovado.");
  } else {
    console.log("Aluno reprovado.");
  }
  console.log(`Media final: ${novaMedia.toFixed(1)}`);
}

main();
